x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/Match.java

```diff
@@ -52,10 +80,14 @@ public class Match extends FullTextFunction implements Validatable {
     )
     public Match(
         Source source,
-        @Param(name = "field", type = { "keyword", "text" }, description = "Field that the query will target.") Expression field,
+        @Param(
+            name = "field",
+            type = { "keyword", "text", "boolean", "date", "date_nanos", "double", "integer", "ip", "long", "unsigned_long", "version" },
+            description = "Field that the query will target."
+        ) Expression field,
         @Param(
             name = "query",
-            type = { "keyword", "text" },
```

Comment:
As a side note I am wondering if it's ever possible to have the query param be a `text` field - the reason for this is that all string literals in ES|QL are treated as `keyword` and we don't have a `to_text` conversion function.
Because we need the query field to be a constant, it would never be a mapped field in an index, which is the only time we are dealing with `text` columns in ES|QL? So could we potentially remove `text` as an allows param type for `query`?

Reply:
I've removed TEXT from the query types, as it can't be actually used.